### PR TITLE
Added cachestat tool

### PR DESCRIPTION
**Pull request by mcaleavya**
Added cachestat tool, man page and exmaple file.

### CODE REVIEW

File: tools/cachestat_example [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# ./cachestat -h
```
Comment (4ast) [inline tools/cachestat_example]: this file is a duplicate of .txt. please remove it. Other than that, it looks great! Awesome tool. Thanks

**Comment by mcaleavya** [comment]
I have removed the duplicate file

**Comment by 4ast** [comment]
not sure why build checks are failing. could you squash all three commits into one and retry?

**Comment by 4ast** [comment]
thanks!

**Pull request merged**